packages/block-library/src/cover/save.js

```diff
@@ -88,7 +88,8 @@ export default function save( { attributes } ) {
 			'has-custom-content-position':
 				! isContentPositionCenter( contentPosition ),
 		},
-		getPositionClassName( contentPosition )
+		getPositionClassName( contentPosition ),
+		'has-no-zindex-in-inner-blocks'
```

Comment:
I didn't like this name, but I couldn't find a better one. I though about a `has-natural-elements-order` or something like this, but the `has-no-zindex-in-inner-blocks` seems more self-explanatory about the purpose of this.

Reply:
Please, let me know if I'm missing something from your suggestion, but I think we still need it.

The main goal of our change is the `z-index` cleanup, removing especially the z-index from the `.wp-block-cover__inner-container`. But we can do it only with the new markup, otherwise, it would break styles for those that don't re-save their posts.

In other words, the `has-no-zindex-in-inner-blocks` class is more like a versioning class (with a more semantic name than a `vXX`) to check if we have the new markup to apply the new styles.

Reply:
>  removing especially the z-index from the .wp-block-cover__inner-container. But we can do it only with the new markup, otherwise, it would break styles for those that don't re-save their posts.

The container could have a style using `:has()` that checks the order of direct children using the `+` sibling selector. If they are in the old order, the style applies the z-index for BC. If the children are in the new order, no z-index would be applied. Another option in our toolkit would be to use `::first-child` when checking the order etc.

As noted in my [other comment](https://github.com/WordPress/gutenberg/pull/66249#discussion_r1812043998), I might not have the bandwidth to give this a try until later this week or early next. Hopefully, this makes sense and gives you a thread to follow though 🧵

Reply:
Yes! I could iterate and answered in the other thread. 😉 Thank you for your suggestions!
https://github.com/WordPress/gutenberg/pull/66249#discussion_r1812826334